### PR TITLE
feat(core): vectorize a swept body that returns a batch

### LINKED ISSUE #405: Vectorize sweep bodies that return a batch, instead of falling back to sequential dispatch
### Current behavior

A sweep whose body returns a `Batch` fails the JAX trace probe in `Node._jax_traceability_error` and falls back to sequential dispatch. That is correct — the two dispatch paths agree on results by contract — but it forgoes vectorization for a whole class of bodies.

### Why the fallback exists

When `vmap` stacks the body's results it inserts an output axis and re-invokes `RecordBatch`'s pytree unflatten hook. The newly created axis has no level name, and the pytree contract gives the hook no way to learn one: the hook receives the static aux replayed from the treedef plus the transformed children, and neither `in_axes` nor `out_axes` is part of that. Without a channel, the reconstructed batch cannot state an honest `BatchSpec` for the axis `vmap` just added.

### Proposal

Add a private output adapter to the workflow executor. Instead of returning a `RecordBatch` from the mapped body, return a private pytree payload whose children are the returned batch's raw columns and whose aux carries:

- the returned batch's inner `BatchSpec`;
- the sweep's own level names and axis groups;
- the `out_axes` placement.

Its unflatten hook then builds the final batch explicitly, with levels ordered as the sweep's followed by the body's.

This puts the reconstruction where the metadata already is. `execute_sweep_rows_jax` knows the sweep's levels and axis groups, and already constructs each body argument explicitly — it flattens each batched input to raw leaf columns and rebuilds a `Record` inside the traced call — so the input side is already free of inference. The adapter does the same for the output side.

### Non-goals

- **Not a public API.** A public level-aware mapping method could later wrap this executor, but it is not required, and one that corrected only the input side would leave the output axis unnamed.
- **Not `jax.custom_batching.custom_vmap`**, which customizes one function's batching rule rather than pytree reconstruction, and does not on its own support reverse-mode differentiation.
- **Not JAX's experimental `hijax` mapping specs.** Those could carry level identity through `vmap`, but only by redesigning `RecordBatch` as a custom JAX type with its own tracing, batching, lowering, and autodiff behavior.

### Acceptance

A sweep whose body returns a batch runs under `vmap` and yields a batch whose levels are the sweep's followed by the body's, with `batch_shape` and `batch_size` consistent with its columns, and whose result matches sequential dispatch.

### Depends on

The `RecordBatch` stack — #399 introduces the class, #402 the sweep executor this would extend.

This is performance-only. The current fallback is correct, so it does not block that stack.

### PR DESCRIPTION
**Pull request by jhuggins**
## Summary

A sweep whose body returns a `RecordBatch` now runs under `jax.vmap` instead of
dropping to row-wise dispatch.

`vmap` inserts an output axis and re-enters `RecordBatch`'s unflatten hook, which
has no name to give the new level: a treedef replays the static aux and the
transformed children, and carries neither `in_axes` nor `out_axes`. **A shape is
not a provenance**, so the hook refuses rather than guess — and that refusal stays
exactly as it is.

What changes is that the executor stops routing through it. The body's returned
batch is taken apart into `_MappedBatchColumns`, an inert private carrier whose
own unflatten rebuilds it verbatim and checks nothing, which is what lets it cross
a rank change the batch cannot. The executor rebuilds the batch on the far side,
where it holds the level names the hook lacked.

This is the move the input side already makes: `execute_sweep_rows_jax` flattens
each batched argument to raw leaf columns and rebuilds a `Record` inside the
traced call, so nothing infers a batch axis on the way in. Now the same holds on
the way out.

## Structure

- **The probe and the executor share one body wrapper**
  (`_workflow_sweep.mapped_row_body`) rather than being two functions kept alike
  by a comment. That correspondence is what drifted to produce #408's bug, so it
  is now structural.
- **One reconstruction serves both dispatches.** `_batch_over_swept_columns`
  builds the aggregate: the row-wise path calls it after stacking rows itself,
  the mapped path with the transform's output axis already in place. The two
  agree by construction instead of by parallel maintenance.

## Linked issue(s)

Closes #405.

Stacked on #408, which fixed the case where the same class of body did not fall
back but crashed. This PR targets that branch and will retarget to `main` when it
merges.

## Test plan

`tests/core/test_workflow_sweep.py`, new class `TestASweptBodyThatReturnsABatch`
— 9 tests:

- **The sweep takes the mapped path** — the discriminating one, verified
  before/after by spying on `execute_sweep_rows_jax`: on #408's head it is
  reached 0 times and the log says "not JAX-traceable; using sequential
  dispatch"; here it is reached once, with no fallback.
- **The levels are the sweep's followed by the body's** — `('row', 'k')`, groups
  `((4,), (3,))`.
- **The shape agrees with the columns it holds** — `batch_shape=(4, 3)`,
  `batch_size=12`, column shape `(4, 3)`, since a batch whose spec its own
  columns contradict is the failure to avoid.
- **It matches sequential dispatch** — on element spec, levels, groups, and
  values.
- **A multi-axis level reshapes on the mapped path** — one argument carrying a
  level over two axes (`batch_shape=(2, 3)`), which is what exercises the
  carrier's flatten-to-`prod`-and-restore. It takes a single argument on
  purpose: two array arguments make two zip groups, which sets `jax_supported`
  false, so the earlier two-argument version measured row-wise dispatch and left
  this uncovered. A spy asserts the mapped executor ran.
- **Two zip groups sweep as a product** — `('a', 'b', 'k')` over
  `((2,), (3,), (2,))`. Row-wise by construction, so it pins the aggregation
  rather than the mapped path.
- **A nested element survives**, since columns are leaf-keyed.
- **The carrier does not reach the caller.**
- **A raw `jax.vmap` returning a batch is still refused** — the hook is routed
  around, not softened, which #406 depends on.

Full suite: 4621 passed, 54 skipped. `ruff check .` and `ruff format --check .`
clean from the repo root.

## Breaking changes

None. Calls that previously fell back now vectorize and produce the same result;
the two dispatches agree by contract, and the tests assert that agreement
directly rather than approximating it.

## Documentation

CHANGELOG entry under `Unreleased → Added`. `_MappedBatchColumns` carries a
docstring stating why it is inert and why that is the other half of the hook's
refusal rather than an exception to it; `mapped_row_body` states why both callers
must use it.

- [x] Docstrings updated for changed public APIs
- [ ] User Guide / tutorials updated where relevant
- [x] CHANGELOG (or release-notes entry) noted in the linked issue

## Checklist

- [x] PR title follows `<type>(<scope>): <subject>`.
- [x] At least one `area:*` label is applied.
- [x] Linked to the primary implementation issue above.

### CODE REVIEW

File: tests/core/test_workflow_sweep.py [hunk 1]
```diff
@@ -268,3 +273,140 @@ def distribution_broadcast(
...
+    def test_it_matches_sequential_dispatch(self):
+        mapped = Function(func=self._body, name="swept")(self._rows(4))
+        sequential = Function(func=self._body, name="swept", dispatch="sequential")(self._rows(4))
+
+        assert mapped.element_spec == sequential.element_spec
+        assert mapped.level_names == sequential.level_names
+        assert mapped.axis_groups == sequential.axis_groups
+        np.testing.assert_allclose(np.asarray(mapped["y"]), np.asarray(sequential["y"]))
+
+    def test_a_multi_axis_sweep_keeps_the_groups_apart(self):
```
Comment (SoymilkWinsAgain) [inline tests/core/test_workflow_sweep.py]: This test doesn't actually go through the JAX path. It takes two arguments and two array groups as input, and the executor explicitly sets `jax_supported=False` in this case. Therefore, the result named `mapped` actually comes from sequential dispatch and doesn't cover the reshape and level reconstruction of the new carrier on the multi-axis outer sweep.

The AI ​​suggests adding a single-parameter, single-level `RecordBatch` with `batch_shape=(2, 3)` and `axis_groups=((2, 3),)`, and then using `spy execute_sweep_rows_jax` to verify the final levels, shape, columns, and sequential equivalence.

**Review by SoymilkWinsAgain — DISMISSED** [review]
There's a small issue in `test_workflow_sweep.py`, no fundamental issues. It is recommended to fix and then merge
